### ISSUE TITLE
Supervisor HTTP not responding

### ISSUE THREAD

**Issue by laneyweb** [lloesche/valheim-server-docker#206]
Hi, I have tried to connect to the 9001 port for the Supervisor function. I'm hitting it from a browser that can resolve to the Valheim server IP and play fine. I cant get the IP:9001 to resolve though. I've added the SUPERVISOR_HTTP and SUPERVISOR_HTTP_PASS as env variables in the docker setup. Is there any tips or checks I can do please?

Thanks!

**Comment by laneyweb** [comment]
Thank you for the quick reply..

I had indeed forgotten the expose 9001. I have fixed that now and the port is exposed. 

From another client, I get this when I curl:
```
* Expire in 0 ms for 6 (transfer 0x20098b0)
*   Trying 192.168.1.99...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x20098b0)
* Connected to 192.168.1.99 (192.168.1.99) port 9001 (#0)
> GET / HTTP/1.1
> Host: 192.168.1.99:9001
> User-Agent: curl/7.64.0
> Accept: */*
>
* Recv failure: Connection reset by peer
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer
```

Out of interest, I cant run curl from the shell of the container, I get 
``` 
root@lloesche-valheim-server1:/# curl
bash: curl: command not found
```
In the logs, I have this but no the "Included extra file" line you mention
```2021-03-10 17:28:50,257 INFO waiting for valheim-updater, valheim-backup to die
2021-03-10 17:28:51,259 INFO stopped: valheim-backup (terminated by SIGTERM)
2021-03-10 17:28:51,260 INFO stopped: valheim-updater (terminated by SIGTERM)
2021-03-10 17:29:38,002 INFO Set uid to user 0 succeeded
2021-03-10 17:29:38,018 INFO RPC interface 'supervisor' initialized
2021-03-10 17:29:38,018 INFO supervisord started with pid 1
2021-03-10 17:29:39,021 INFO spawned: 'valheim-updater' with pid 8
2021-03-10 17:29:39,023 INFO spawned: 'valheim-backup' with pid 9`
```
And the ps command shows this:
```
root@lloesche-valheim-server1:/# ps
  PID TTY          TIME CMD
  324 pts/1    00:00:00 bash
  330 pts/1    00:00:00 ps
root@lloesche-valheim-server1:/# ps -x
  PID TTY      STAT   TIME COMMAND
    1 pts/0    Ss+    0:00 /usr/bin/python2 /usr/bin/supervisord -c /etc/supervisor/supervisord.conf
    8 pts/0    S      0:00 /bin/bash /usr/local/bin/valheim-updater
    9 pts/0    S      0:00 /bin/bash /usr/local/bin/valheim-backup
   42 pts/0    S      0:00 sleep 3600
   52 pts/0    S      0:00 /bin/bash /usr/local/bin/valheim-server
   53 pts/0    Sl     3:52 /opt/valheim/valheim_server.x86_64 -nographics -batchmode -name * -port 2456 -world * -password /* -public 1
  324 pts/1    Ss     0:00 bash
  332 pts/0    S      0:00 sleep 5
  333 pts/1    R+     0:00 ps -x
```

**Comment by laneyweb** [comment]
just did an apt-get for curl on the container so I could run there and I get this back:

```
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Expire in 149999 ms for 3 (transfer 0x556601cd9f90)
* Expire in 200 ms for 4 (transfer 0x556601cd9f90)
* connect to 127.0.0.1 port 9001 failed: Connection refused
*   Trying ::1...
* TCP_NODELAY set
* Expire in 149999 ms for 3 (transfer 0x556601cd9f90)
* Immediate connect fail for ::1: Cannot assign requested address
*   Trying ::1...
* TCP_NODELAY set
* Expire in 149999 ms for 3 (transfer 0x556601cd9f90)
* Immediate connect fail for ::1: Cannot assign requested address
* Failed to connect to localhost port 9001: Connection refused
* Closing connection 0
curl: (7) Failed to connect to localhost port 9001: Connection refused
```

**Comment by lloesche** [comment]
> When you're inside the container can you curl -v localhost:9001? This should return a 401 Unauthorized.

**Comment by lloesche** [comment]
Also, make sure to pull the latest container image... just in case you're still running on a version that doesn't have the supervisor http server.

**Comment by lloesche** [comment]
> just did an apt-get for curl on the container so I could run there and I get this back:
> curl: (7) Failed to connect to localhost port 9001: Connection refused
> ```

There is no need to `apt-get install curl` inside the container. `curl` ships with the container. Wherever you executed this it was not inside the container. Maybe you were on the container host?

**Comment by laneyweb** [comment]
OK. Let me take another look at my setup, there is something off and I may be the cause by not fully understanding the docker setup.

**Comment by lloesche** [comment]
> OK. Let me take another look at my setup, there is something off and I may be the cause by not fully understanding the docker setup.

No worries we'll figure it out.

The idea is to get a shell inside the container.

If you're starting it with `docker run --name valheim-server` you'd get a shell using
```
$ docker exec -it valheim-server /bin/bash
```

If you're running in K8S the command would be
```
kubectl exec --stdin --tty valheim-server -- /bin/bash
```

**Comment by laneyweb** [comment]
I am not using Docker command line. I have a Synology so I am using their interface. I also use Portainer for managing the containers when they are running. I have used Portainer and the Synology Docker UI and both show the container to have no curl installed. I'm pretty sure I am using the Synology and Portainer tools to get to the container shell. I'm certainly not looking at the host/synology shell. that has curl etc.

I removed the container and grabbed the latest build today but now I am unable to start the container at all - I have posted a reply to https://github.com/lloesche/valheim-server-docker/issues/208 this thread where someone else seems to have the same issue.
